package.json

```diff
@@ -97,6 +97,7 @@
     "react-native-logs": "^5.1.0",
     "react-native-mail": "github:chirag04/react-native-mail",
     "react-native-maps": "^1.13.0",
+    "react-native-mmkv": "^2.12.2",
```

Comment:
A couple questions on this:

1. Do we need a key/value storage solution for Settings, or can we store this information in Realm along with all our other local data? We already have a [LocalPreferences](https://github.com/inaturalist/iNaturalistReactNative/blob/main/src/realmModels/LocalPreferences.js) model in Realm which could include this data without the complexity of introducing a new library.
2. If you're committed to using `react-native-mmkv` here, can you make a separate issue to replace AsyncStorage in the rest of the app? Right now, we're using 4 different libraries for local storage (`mmkv, async-storage, realm, and react-native-sensitive-info`) which creates developer confusion on which library to use and what data we're storing. I think it's better for us to make a clear decision on which storage solution(s) we really need and remove the rest. This way we have more of a single source of truth for local data.

Reply:
1.) We do not need a key/value storage for this setting. What we do need is to persist it between app launches. We can store it in Realm. I did not see the LocalPreferences before, thanks for mentioning it. I do think though that the overhead for using Realm is quite large for this specific boolean. So in terms of complexity it seems more complex to add it to Realm than a key/value storage. It also feels weird to use Realm which I think of more as being for collections for one item only.
2.) I am not commited to this dependency, so I will revert to another solution.
My main concern here was speed of retrieval of the value, as it is a value we need at app start, and everything that makes the app taking longer to start makes for a worse UX and higher churn.
But then my question becomes, why do we have three different storages already, plus `zustand` for app state?

Reply:
Sounds good. To be clear, I'm not being prescriptive one way or another -- mmkv might make the most sense here if it's performant. I'm just feeling confused about why we have so many different sources of truth in the app & would like to reduce developer confusion if possible. Afaik the only reason we have three different storages is because we each have different developer preferences on which libraries we prefer to use -- it would make sense to me to pick one if possible.

`zustand` serves a different purpose since it preserves app state while a user is in the app, while the storage solutions preserve data between app launches.